ext/exif/exif.c

```diff
@@ -2025,6 +2029,65 @@ typedef struct {
 } jpeg_sof_info;
 /* }}} */
 
+/* Base address for offset references, together with valid memory range.
+ * The valid range does not necessarily include the offset base. */
+typedef struct {
+	char *offset_base;
+	char *valid_start; /* inclusive */
+	char *valid_end;   /* exclusive */
+} exif_offset_info;
```

Comment:
I like this approach but I wonder whether it won't be more comprehensive (even though more time-consuming) to modify various fetch functions like php_ifd_get32u to take struct instead of naked pointer, and thus ensure there's never an out-of-buffer access no matter what?

Reply:
I generally agree, but this would require a much larger patch, so I would prefer to land this reduced variant first.